File: editor/en/[1431]Kids With the Greatest Number of Candies.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = 0;
        for (int num : candies) {
            max = Math.max(max, num);
        }
        List<Boolean> list = new ArrayList<>();
        max -= extraCandies;
        for (int num : candies) {
            list.add(num >= max);
        }
        return list;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[1431]Kids With the Greatest Number of Candies.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        max_num = max(candies)
        return [candy + extraCandies >= max_num for candy in candies]
# leetcode submit region end(Prohibit modification and deletion)
